cognite/neat/_data_model/deployer/data_classes.py

```diff
@@ -152,18 +168,21 @@ class SchemaSnapshot(BaseDeployObject):
 
 class ChangeResult(BaseDeployObject, Generic[T_ResourceId, T_DataModelResource]):
     change: ResourceChange[T_ResourceId, T_DataModelResource]
-    message: ItemMessage[T_ResourceId]
+    message: SuccessResponseItems[T_ResourceId] | FailedResponseItems[T_ResourceId] | FailedRequestItems[T_ResourceId]
 
 
-class AppliedChanges(BaseDeployObject, Generic[T_ResourceId, T_DataModelResource]):
-    created: list[ChangeResult[T_ResourceId, T_DataModelResource]] = Field(default_factory=list)
-    updated: list[ChangeResult[T_ResourceId, T_DataModelResource]] = Field(default_factory=list)
-    deletions: list[ChangeResult[T_ResourceId, T_DataModelResource]] = Field(default_factory=list)
-    unchanged: list[ChangeResult[T_ResourceId, T_DataModelResource]] = Field(default_factory=list)
+class AppliedChanges(BaseDeployObject):
```

Comment:
We cannot have generic here, as the list can be ChangeResult of different types.